sampler.go

```diff
@@ -438,7 +477,23 @@ func (s *RemotelyControlledSampler) updateSampler() {
 }
 
 func (s *RemotelyControlledSampler) extractSampler(res *sampling.SamplingStrategyResponse) (Sampler, error) {
-	// TODO create adaptive sampler
+	if strategies := res.GetOperationSamplingStrategies(); strategies != nil {
+		samplingRates := make(map[string]float64)
+		for _, strategy := range strategies {
+			samplingRates[strategy.Operation] = strategy.ProbabilisticSampling.SamplingRate
+		}
+		if sampler, ok := s.sampler.(*adaptiveSampler); ok {
+			s.Lock()
+			sampler.update(defaultLowerBound, defaultSamplingProbability, samplingRates)
+			s.Unlock()
```

Comment:
this is not good design-wise, you just added side effects to a previously referentially transparent function. I'd say just return a new adaptive sampler here, but handle the intelligent update in `updateSampler()`, which already holds the locks.